src/libcore/fmt/builders.rs

```diff
@@ -1,37 +1,50 @@
 use crate::fmt;
 
-struct PadAdapter<'a> {
-    buf: &'a mut (dyn fmt::Write + 'a),
+struct PadAdapter<'buf, 'state> {
+    buf: &'buf mut (dyn fmt::Write + 'buf),
+    state: &'state mut PadAdapterState,
```

Comment:
Shifting the state of the `PadAdapter` into a dedicated struct here lets `value` pick up where `key` left off without having to guess what the state was before.